core/src/providers/openai.rs

```diff
@@ -1200,12 +1200,17 @@ impl Embedder for OpenAIEmbedder {
             ));
         }
 
-        // Give priority to `CORE_DATA_SOURCES_OPENAI_API_KEY` env variable
```

Comment:
This whole change is about making sure that if `CORE_DATA_SOURCES_OPENAI_API_KEY` is set, we don't end up with `use_eu_endpoint` set to true.